File: 3132. Find the Integer Added to Array II/Solution.cpp
#include <ranges>
#include <vector>
using namespace std;

class Solution {
public:
    int minimumAddedInteger(vector<int>& nums1, vector<int>& nums2)
    {
        ranges::sort(nums1);
        ranges::sort(nums2);
        for (int i = 2; i > 0; i--) {
            int d = nums2[0] - nums1[i];
            for (size_t j = i, k = 0; j < nums1.size(); j++)
                if (nums1[j] + d == nums2[k] && ++k == nums2.size())
                    return d;
        }
        return nums2[0] - nums1[0];
    }
};

File: 3132. Find the Integer Added to Array II/Solution.py
from typing import List


class Solution:
    def minimumAddedInteger(self, nums1: List[int], nums2: List[int]) -> int:
        nums1.sort()
        nums2.sort()
        for i in (2, 1):
            d = nums2[0] - nums1[i]
            it = iter(nums1)
            if all(x - d in it for x in nums2):
                return d
        return nums2[0] - nums1[0]
